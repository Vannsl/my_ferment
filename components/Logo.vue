<template>
  <a
    :class="classList"
    class="no-underline hover:no-underline font-bold text-2xl lg:text-4xl"
    href="#"
  >
    <logo-svg />&nbsp;MyFerments
  </a>
</template>

<script>
import LogoSvg from '@/components/svg/logo'

export default {
  name: 'Logo',
  components: {
    'logo-svg': LogoSvg
  },
  props: {
    isStickable: {
      type: Boolean,
      default: false
    },
    isSticky: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    classList() {
      let classList = ''
      if (this.isStickable) {
        classList = this.isSticky ? 'text-gray-800' : 'text-white'
      } else {
        classList = 'text-primary'
      }
      return classList
    }
  }
}
</script>