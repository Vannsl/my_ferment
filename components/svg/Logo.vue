<template>
  <!--<div>Icons made by <a href="https://www.flaticon.com/authors/freepik" title="Freepik">Freepik</a> from <a href="https://www.flaticon.com/"             title="Flaticon">www.flaticon.com</a></div> -->
  <svg
    class="h-8 fill-current inline"
    version="1.1"
    id="Capa_1"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:xlink="http://www.w3.org/1999/xlink"
    x="0px"
    y="0px"
    viewBox="0 0 382.421 382.422"
    xml:space="preserve"
  >
    <g>
      <path
        d="M297.209,80.934V51.36H83.347v28.365c-22.312,14.323-33.62,37.431-33.62,68.758v145.688
		c0,58.56,40.534,88.251,120.473,88.251h42.023c79.936,0,120.472-29.691,120.472-88.251V148.482
		C332.695,118.258,320.76,95.56,297.209,80.934z M315.885,294.175c0,48.076-33.905,71.442-103.662,71.442H170.2
		c-69.757,0-103.662-23.366-103.662-71.442V148.489c0-27.115,9.588-45.651,29.311-56.667l4.309-2.405V68.17H280.4v22.569
		l4.371,2.394c20.937,11.453,31.114,29.558,31.114,55.35V294.175z M78.098,27.316V11.207C78.098,5.021,83.118,0,89.304,0h203.819
		c6.187,0,11.206,5.021,11.206,11.207v16.109c0,6.187-5.02,11.207-11.206,11.207H89.304C83.113,38.522,78.098,33.502,78.098,27.316z
		 M77.744,177.591H304.68v91.051H77.744V177.591z"
      />
    </g>
  </svg>
</template>

<script>
export default {
  name: 'LogoSvg'
}
</script>