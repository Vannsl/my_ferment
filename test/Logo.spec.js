import { shallowMount } from '@vue/test-utils'
import Logo from '@/components/Logo'

const factory = () => {
  return shallowMount(Logo, {
    propsData: {
      isStickable: true,
      isSticky: false
    }
  })
}

describe('Logo', () => {
  it('is a Vue instance', () => {
    const wrapper = factory()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders properly', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('has the property isStickable', () => {
    const wrapper = factory()
    expect(wrapper.props('isStickable')).toBe(true)
  })

  it('has the property isSticky', () => {
    const wrapper = factory()
    expect(wrapper.props('isSticky')).toBe(false)
  })

  describe('when creating the classlist', () => {
    describe('when isStickable is true and isSticky is false', () => {
      it('has the class', () => {
        const wrapper = factory()
        const anchor = wrapper.find('a')
        expect(anchor.classes('text-white')).toBe(true)
      })
    })

    describe('when isStickable is true and isSticky is true', () => {
      it('has the class', () => {
        const wrapper = factory()
        wrapper.setProps({ isSticky: true })
        const anchor = wrapper.find('a')
        expect(anchor.classes('text-gray-800')).toBe(true)
      })
    })

    describe('when isStickable is false', () => {
      it('has the class', () => {
        const wrapper = factory()
        wrapper.setProps({ isStickable: false })
        const anchor = wrapper.find('a')
        expect(anchor.classes('text-orange-600')).toBe(true)
      })
    })
  })
})
