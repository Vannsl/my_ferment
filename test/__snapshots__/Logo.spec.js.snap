// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders properly 1`] = `
"<a href=\\"#\\" class=\\"no-underline hover:no-underline font-bold text-2xl lg:text-4xl text-white\\"><logo-svg-stub></logo-svg-stub>&nbsp;LANDING
</a>"
`;
